diff --git a/blocks.def.h b/blocks.def.h
index dbb3305..5907c68 100644
--- a/blocks.def.h
+++ b/blocks.def.h
@@ -1,11 +1,15 @@
+#define BIN_PATH "~/.local/bin/statusbar/"
+
 //Modify this file to change what commands output to your statusbar, and recompile using the make command.
 static const Block blocks[] = {
 	/*Icon*/	/*Command*/		/*Update Interval*/	/*Update Signal*/
-	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g",	30,		0},
-
-	{"", "date '+%b %d (%a) %I:%M%p'",					5,		0},
+	{"", BIN_PATH "sb-memory",					5,		5},
+	{"", BIN_PATH "sb-internet",				5,		4},
+	{"", BIN_PATH "sb-volume",					5,		3},
+	{"", BIN_PATH "sb-battery",				5,	   2},
+	{"", BIN_PATH "sb-date",					5,		1},
 };
 
-//sets delimiter between status commands. NULL character ('\0') means no delimiter.
+//sets delimeter between status commands. NULL character ('\0') means no delimeter.
 static char delim[] = " | ";
 static unsigned int delimLen = 5;
