--- blocks.def.h.origin	2025-07-04 00:31:29.697312161 +0200
+++ blocks.def.h	2025-07-03 20:46:25.975273631 +0200
@@ -1,11 +1,14 @@
+#define BIN_PATH "~/.local/bin/statusbar/"
+
 //Modify this file to change what commands output to your statusbar, and recompile using the make command.
 static const Block blocks[] = {
 	/*Icon*/	/*Command*/		/*Update Interval*/	/*Update Signal*/
-	{"Mem:", "free -h | awk '/^Mem/ { print $3\"/\"$2 }' | sed s/i//g",	30,		0},
-
-	{"", "date '+%b %d (%a) %I:%M%p'",					5,		0},
+	{"", BIN_PATH "sb-memory",					5,		4},
+	{"", BIN_PATH "sb-volume",					5,		3},
+	{"", BIN_PATH "sb-battery",				5,	   2},
+	{"", BIN_PATH "sb-date",					5,		1},
 };
 
-//sets delimiter between status commands. NULL character ('\0') means no delimiter.
+//sets delimeter between status commands. NULL character ('\0') means no delimeter.
 static char delim[] = " | ";
 static unsigned int delimLen = 5;
